### PR TITLE
Use file scoped namespaces in rspec template

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Cleanup after #5590 

Changes:
* Removed `DiagnosticDescriptorBuilder.GetDescriptor` - will be deleted
* Using file scope namespaces
* `SonarDiagnosticAnalyzer<TSyntaxKind>` base class to lower boiler plate noise

### CODE REVIEW

File: scripts/rspec/rspec.ps1 [hunk 1]
```diff
@@ -173,29 +173,28 @@ function GenerateBaseClassIfSecondLanguage()
...
         $existingCSClassText = ReplaceTextInString -oldText $oldCsClass -newText $newCsClass -modifiableString $existingCSClassText
         $existingVBClassText = ReplaceTextInString -oldText $oldVbClass -newText $newVbClass -modifiableString $existingVBClassText
 
-        $diagnosticIdToken = "        private const string DiagnosticId = ""${ruleKey}"";"
+        $diagnosticIdToken = "    private const string DiagnosticId = ""${ruleKey}"";"
         $existingCSClassText = RemoveText -textToRemove $diagnosticIdToken -modifiableString $existingCSClassText
         $existingVBClassText = RemoveText -textToRemove $diagnosticIdToken -modifiableString $existingVBClassText
 
-        $messageFormatToken = "       private const string MessageFormat = """";"
+        $messageFormatToken = "   private const string MessageFormat = ""FIXME"";"
```
Comment (csaba-sagi-sonarsource) [inline scripts/rspec/rspec.ps1:187]: Educational: Why did you add the FIXME there?

Comment (pavel-mikula-sonarsource) [reply scripts/rspec/rspec.ps1:187]: So when we add a new rule, we can more clearly see that we have to change that
